### PR TITLE
Example refactor and Funding Source models

### PR DESCRIPTION
**Pull request by ithielnor**
Added funding source models.
Added some basic inheritance to Response models
Refactored example app into separate classes

This PR depends on #5

### CODE REVIEW

**Comment by ithielnor** [comment]
This now includes a breaking change on CreateCustomerRequest.

**Comment by ithielnor** [comment]
Hey @therockstorm, Anything I can do to move this PR along?

**Comment by therockstorm** [comment]
@ithielnor I haven't had a chance to look at it, I've been busy trying to role new functionality to our API. Hopefully, I get to it early next week.

File: Dwolla.Client/Models/Responses/BasegetResponse.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
```
Comment (therockstorm) [inline Dwolla.Client/Models/Responses/BasegetResponse.cs]: Filename should have capital 'g' in "Get"

File: Dwolla.Client/Models/Responses/BusinessClassification.cs [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
-    public class IndustryClassificationsEmbed
+    public class IndustryClassificationsEmbed : Embed<IndustryClassification>
     {
         [JsonProperty(PropertyName = "industry-classifications")]
         public List<IndustryClassification> IndustryClassifications { get; set; }
+
+        public override List<IndustryClassification> Results()
+        {
+            return IndustryClassifications;
+        }
```
Comment (therockstorm) [inline Dwolla.Client/Models/Responses/BusinessClassification.cs:24]: This (and the other Results) can be expression-bodied,
`public override List<IndustryClassification> Results() => IndustryClassifications;`

File: Dwolla.Client/Models/Responses/FundingSource.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public class FundingSource : BaseResponse
+    {
+        public string Id { get; set; }
+        public string Status { get; set; }
+        public string Type { get; set; }
+        public string BankAccountType { get; set; }
+        public string Name { get; set; }
+        public DateTime Created { get; set; }
+        public bool Removed { get; set; }
+        public string[] Channels { get; set; }
```
Comment (therockstorm) [inline Dwolla.Client/Models/Responses/FundingSource.cs]: Why not use `List<string>` (same for the other arrays)?

File: ExampleApp/DwollaBroker.cs [hunk 4]
```diff
@@ -32,19 +32,34 @@ public class DwollaBroker
         public async Task<RootResponse> GetRootAsync() =>
             (await GetAsync<RootResponse>(new Uri(_client.ApiBaseAddress))).Content;
 
-        public async Task<Uri> CreateCustomerAsync(Uri uri, string firstName, string lastName, string email)
+        public async Task<Uri> CreateCustomerAsync(Uri uri, string firstName, string lastName, string email, DateTime? dateOfBirth = null)
```
Comment (therockstorm) [inline ExampleApp/DwollaBroker.cs]: Dob is only required for VCRs but so are other fields that are shown here. I'd rather not add it to the example and cause confusion.

File: ExampleApp/Tasks/BaseTask.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        protected static string RandomNumberString(int length)
+        {
+            const string chars = "0123456789";
+            return new string(Enumerable.Repeat(chars, length).Select(s => s[Random.Next(s.Length)]).ToArray());
+        }
+
+        protected static string RandomString(int length)
+        {
+            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+            return new string(Enumerable.Repeat(chars, length).Select(s => s[Random.Next(s.Length)]).ToArray());
```
Comment (therockstorm) [inline ExampleApp/Tasks/BaseTask.cs]: Can be extracted to a method that takes the chars string so both Random... methods can use it

File: ExampleApp/Program.cs [hunk 6]
```diff
@@ -36,104 +39,60 @@ static void Main()
...
-                RandomString(10));
-
-            var subscription = await broker.GetWebhookSubscriptionAsync(createdSubscriptionUri);
-            WriteLine($"Created Subscription {subscription.Id} with url={subscription.Url}");
-
-            return subscription;
+            WriteLine(@"Options:
+ - Quit (q)
+ - Help (?)");
+            GetTasks().ForEach(ta => WriteLine($" - {ta.Description} ({ta.Command})"));
```
Comment (therockstorm) [inline ExampleApp/Program.cs:64]: This is pretty slick! 👍

**Review by therockstorm — CHANGES_REQUESTED** [review]

File: Dwolla.Client/Models/Responses/BasegetResponse.cs (hunk 1 above)
Comment (ithielnor) [reply Dwolla.Client/Models/Responses/BasegetResponse.cs]: Had to do this in two commits cause github wouldn't take it.

File: Dwolla.Client/Models/Responses/FundingSource.cs (hunk 3 above)
Comment (ithielnor) [reply Dwolla.Client/Models/Responses/FundingSource.cs]: `List<string>` implies `Add()` and similar methods in my mind. I can switch to `List<>` if that's what you'd prefer.

File: ExampleApp/DwollaBroker.cs (hunk 4 above)
Comment (ithielnor) [reply ExampleApp/DwollaBroker.cs]: I think I added this to test out some details of verification. Removed.

**Pull request merged**